source/common/http/utility.cc

```diff
@@ -217,6 +218,10 @@ initializeAndValidateOptions(const envoy::config::core::v3::Http2ProtocolOptions
     options_clone.mutable_max_inbound_window_update_frames_per_data_frame_sent()->set_value(
         OptionsLimits::DEFAULT_MAX_INBOUND_WINDOW_UPDATE_FRAMES_PER_DATA_FRAME_SENT);
   }
+  if (!options_clone.has_max_header_list_size()) {
+    options_clone.mutable_max_header_list_size()->set_value(
```

Comment:
Curious if we should just not set the field by default. Since it's a 32-bit unsigned I think what you have is safe, but this does change existing wire behavior technically.

Reply:
If not set, the value will be 0. a smaller value may reject the request.

Reply:
See my above comment. Is it reasonable to not set value here? We can leave this value decided by envoy HCM, which is now 60KB.